### ISSUE TITLE
Advance Image Transformations and SASS compilation not working on Static Publish and Time Machine

### ISSUE THREAD

**Issue by dsilvam** [dotCMS/core#20707]
[Advance Image Transformations](https://dotcms.com/docs/latest/image-resizing-and-processing#advancedImageApi) and 
[SASS compilation](https://dotcms.com/docs/latest/sass-compiler) not working on Static Publish and Time Machine. 

This is happening since 21.05.1 when [Mockito was replaced by a new Mock class](https://github.com/dotCMS/core/issues/20506). 

Stacktrace:

https://gist.github.com/dsilvam/4b41fd54ad2585a32dbd75d5388a92a3

Reproduced in prerelease 21.07

### QA Steps

1. Create a page using the `Landing Page` template
2. Create a `Code Snippet` widget in the created page with the following code:
```html
<img src="/contentAsset/image/d1f449ec-ad6a-4b59-ab38-98ba2e9c3231/fileAsset/crop_w/250/crop_h/150/crop_x/172/crop_y/167">
```
3. Create a push environment with a **static** endpoint
4. Push-Publish the created page
5. Validate in static folder the `application` and `contentAsset` folders were created with their corresponding content, that is the referenced image file inside the code snippet and the `dotsass` file at `${dotTheme.path}css/styles.dotsass`

**Comment by victoralfaro-dotcms** [comment]
Static content generated successfully:
![image](https://user-images.githubusercontent.com/64026161/127196633-238c4a62-1fbe-4329-a994-35362675331a.png)
Also SCSS was compiled in `styles.dotsass` file (truncated sample):
```
*!
 * Bootstrap v4.1.0 (https://getbootstrap.com/)
 * Copyright 2011-2018 The Bootstrap Authors
 * Copyright 2011-2018 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 */:root{--blue: #007bff;--indigo: #6610f2;--purple: #6f42c1;--pink: #e83e8c;--red: #dc3545;--orange: #fd7e14;--yellow: #ffc107;--green: #28a745;--teal: #20c997;--cyan: #17a2b8;--white: #fff;--gray: #6c757d;--gray-dark: #343a40;--primary: #007bff;--secondary: #6c757d
```

**Comment by victoralfaro-dotcms** [comment]
Page HTML code included with all references:
```html
<!doctype html>
<html lang="en">

    
        

<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>test-dinho</title>
<meta name="description" content="$pDescription">
<meta name="language" content="english">
<meta name="author" content="dotCMS">
<meta name="copyright" content="dotCMS LLC, Miami Florida, US">
<meta property="og:title" content="test-dinho">
<meta property="og:url" content="https://aa.dotcms.io$!{request.getAttribute("javax.servlet.forward.request_uri")}">
<meta property="og:image" content="//dotcms.com/images/default.png">
<link rel="canonical" href="https://aa.dotcms.io$!{request.getAttribute("javax.servlet.forward.request_uri")}">
<link rel="icon" type="image/x-icon" href="/application/themes/landing-page/img/favicon.ico"/>

<!-- CSS -->
<link rel="stylesheet" href="/application/themes/landing-page/css/styles.dotsass">

<!-- Fonts -->
<link rel="dns-prefetch" href="https://fonts.gstatic.com">
<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin="anonymous">
<link rel="preload" href="https://fonts.googleapis.com/css?family=Open+Sans:300,400,400i,600,700,800" as="fetch" crossorigin="anonymous">
<script type="text/javascript">!function(e,n,t){"use strict";var o="https://fonts.googleapis.com/css?family=Open+Sans:300,400,400i,600,700,800",r="__3perf_googleFonts_cf86d";function c(e){(n.head||n.body).appendChild(e)}function a(){var e=n.createElement("link");e.href=o,e.rel="stylesheet",c(e)}function f(e){if(!n.getElementById(r)){var t=n.createElement("style");t.id=r,c(t)}n.getElementById(r).innerHTML=e}e.FontFace&&e.FontFace.prototype.hasOwnProperty("display")?(t[r]&&f(t[r]),fetch(o).then(function(e){return e.text()}).then(function(e){return e.replace(/@font-face {/g,"@font-face{font-display:swap;")}).then(function(e){return t[r]=e}).then(f).catch(a)):a()}(window,document,localStorage);</script>
<!-- End of Google Fonts -->

</head>

<body id="test-dinho" >

    
    <div class="body-wrapper">
    
            
            
                
                        
    
                                                            
                                                
                                                                                                                        
                        <section id="section-1" class="section ">
                            <div class="container">
                                <div class="row">
                        
                                                
                                                                                                
                                    <div class="col-lg-6  ">
                                                                                

<!-- Container Code: /application/themes/landing-page/containers/image.vtl -->
<div class="img-poloroid" style="background-image: url('/dA/90050998-694e-47d8-8ff6-6af1b9127496/services-2.jpg/480w/50q/services-2.jpg');"></div><div></div>
                                    </div><!--/Column-->

                                                                    
                                                
                                                                                                
                                    <div class="col-lg-6  ">
                                                                                
<img src="/contentAsset/image/d1f449ec-ad6a-4b59-ab38-98ba2e9c3231/fileAsset/crop_w/250/crop_h/150/crop_x/172/crop_y/167"><div></div>
                                    </div><!--/Column-->

                                                        </div><!--/row-->
                            </div><!--/container-->
                        </section><!-- /row-wrapper-->
                                                                                                                                                    
                        <section id="section-2" class="section ">
                            <div class="container">
                                <div class="row">
                        
                                                
                                                                                                
                                    <div class="col-lg-6  ">
                                                                                
<div></div>
                                    </div><!--/Column-->

                                                                    
                                                
                                                                                                
                                    <div class="col-lg-6  ">
                                                                                
<div></div>
                                    </div><!--/Column-->

                                                        </div><!--/row-->
                            </div><!--/container-->
                        </section><!-- /row-wrapper-->
                                                                                                                                                    
                        <section id="section-3" class="section ">
                            <div class="container">
                                <div class="row">
                        
                                                
                                                                                                
                                    <div class="col-lg-3  ">
                                                                                
<div></div>
                                    </div><!--/Column-->

                                                                    
                                                
                                                                                                
                                    <div class="col-lg-3  ">
                                                                                
<div></div>
                                    </div><!--/Column-->

                                                                    
                                                
                                                                                                
                                    <div class="col-lg-3  ">
                                                                                
<div></div>
                                    </div><!--/Column-->

                                                                    
                                                
                                                                                                
                                    <div class="col-lg-3  ">
                                                                                
<div></div>
                                    </div><!--/Column-->

                                                        </div><!--/row-->
                            </div><!--/container-->
                        </section><!-- /row-wrapper-->
                                                                                                                                                    
                        <section id="section-4" class="section ">
                            <div class="container">
                                <div class="row">
                        
                                                
                                                                                                
                                    <div class="col-lg-12  ">
                                                                                
<div></div>
                                    </div><!--/Column-->

                                                        </div><!--/row-->
                            </div><!--/container-->
                        </section><!-- /row-wrapper-->
                                                                            
                                                    </div><!-- /body-wrapper -->

            <footer class="footer-wrapper">
    <div class="container">
        <div class="row justify-content-end">
            <div class="col-sm-6">
                <a class="brand" href="/" aria-label="TravelLux Home"><img class="brand-logo-dark" src="/application/themes/landing-page/img/logo.png" width="195" height="40" alt="TravelLux Home"></a>
            </div>
            <div class="col-sm-6">
                <span>&copy;2012 - 2021 All Rights Reserved</span> <a href="https://dotcms.com/company/policies/privacy-policy">Terms of Use</a> <span>and</span> <a href="https://dotcms.com/company/policies/privacy-policy">Privacy Policy</a>
            </div>
        </div>
    </div>
</footer>

<!-- Javascript-->
<script src="/application/themes/landing-page/js/core.min.js"></script>
<script src="/application/themes/landing-page/js/script.js"></script>    
</body>
</html>
```

**Comment by freddyucv** [comment]
it no generated the files for:

<script src="/application/themes/landing-page/js/core.min.js"></script>
<script src="/application/themes/landing-page/js/script.js"></script>    

but We don't kont if it is ok that way

**Comment by bryanboza** [comment]
Fixed, tested on release-21.07 // Postgres // FF

**Comment by swicken-dotcms** [comment]
Not required for 5.3.8.6
